src/passes/GUFA.cpp

```diff
@@ -143,23 +97,11 @@ struct GUFAOptimizer
     auto& wasm = *getModule();
     Builder builder(wasm);
 
-    auto replaceWithUnreachable = [&]() {
-      if (canReplaceWithUnreachable(curr)) {
-        replaceCurrent(getDroppedUnconditionalChildrenAndAppend(
-          curr, wasm, options, builder.makeUnreachable()));
-      } else {
-        // We can't remove this, but we can at least put an unreachable
-        // right after it.
-        replaceCurrent(builder.makeSequence(builder.makeDrop(curr),
-                                            builder.makeUnreachable()));
-      }
-      optimized = true;
```

Comment:
I think maybe the removal of this line is why the test results changed. We are not doing a refinalize like we used to?

I think there might be cases where we need it, so it would be safer to set `optimized=true` in all places we change things.

Reply:
Thanks, done.